python/shotgun_model/shotgun_model.py

```diff
@@ -669,19 +673,87 @@ def _finalize_item(self, item):
         """
         Called whenever an item is fully constructed, either because a shotgun query returned it
         or because it was loaded as part of a cache load from disk.
-        
-        .. note:: You can subclass this if you want to run post processing on 
+
+        .. note:: You can subclass this if you want to run post processing on
             the data as it is arriving. For example, if you are showing a list of
-            task statuses in a filter view, you may want to remember which 
+            task statuses in a filter view, you may want to remember which
             statuses a user had checked and unchecked the last time he was running
             the tool. By subclassing this UI you can easily apply those settings
-            before items appear in the UI. 
+            before items appear in the UI.
 
-        :param item: :class:`~PySide.QtGui.QStandardItem` that is about to be added to the model. 
+        :param item: :class:`~PySide.QtGui.QStandardItem` that is about to be added to the model.
             This has been primed with the standard settings that the ShotgunModel handles.
         """
         # the default implementation does nothing
 
+    def _set_tooltip(self, item, sg_item):
+        """
+        Called when an item is created.
+
+        .. note:: You can subclass this if you want to set your own tooltip for the model item. By
```

Comment:
For reference, this is what this block of code gets rendered like: <img width="653" alt="screen shot 2016-01-12 at 11 10 37 am" src="https://cloud.githubusercontent.com/assets/8126447/12269002/4809a032-b91d-11e5-8175-8d144ac76193.png">